### PR TITLE
Allow for user enter press on input fields

### PR DESCRIPTION
**Pull request by niclim**
## Why
Describe the motivation for the changes.
Allow users to press enter on input fields to go to the next stage (this is usually opening the commit modal)

## What
What's changing? Anything of note to call out?

### Documentation page
pressing enter in one of these pages will open the commit modal
Screen Shot 2021-05-04 at 4 26 38 PM
Screen Shot 2021-05-04 at 4 26 57 PM

Commit modal will submit the message
Screen Shot 2021-05-04 at 4 26 46 PM

### Diff page
Review diffs page - will open the commit modal
*NOTE - I don't actually think this is the right place for contributions (for request / response fields), or even that we should commit here - it may be potentially confusing to have two different workflows for adding contributions? Perhaps having a streamlined diff review process (simplified) where it walks you through what to do? This way a user isn't going to "leave" the diff process half way through, which it feels like currently they can quite easily do so

Screen Shot 2021-05-04 at 4 28 48 PM

Unrecognized url - will stage the endpoint
Screen Shot 2021-05-04 at 4 29 29 PM

Pending + existing endpoints - will open commit modal
Screen Shot 2021-05-04 at 4 29 51 PM

## Validation
* [ ] CI passes
* [ ] Verified in staging
* etc...

### CODE REVIEW

File: workspaces/ui-v2/src/optic-components/pages/diffs/CapturePage.tsx [hunk 1]
```diff
@@ -266,63 +266,6 @@ function DiffCaptureResults() {
   );
 }
 
-/*
```
Comment (niclim) [inline workspaces/ui-v2/src/optic-components/pages/diffs/CapturePage.tsx:269]: This looks like a duplicate of above

File: workspaces/ui-v2/src/optic-components/hooks/edit/Contributions.tsx [hunk 2]
```diff
@@ -106,15 +111,30 @@ export const ContributionEditingStore: FC<ContributionEditingStoreProps> = (
         try {
           await spectacleMutator({
             query: `
-            mutation X($commands: [JSON], $commitMessage: String) {
-              applyCommands(commands: $commands, commitMessage: $commitMessage) {
+            mutation X(
```
Comment (niclim) [inline workspaces/ui-v2/src/optic-components/hooks/edit/Contributions.tsx:114]: One variant of code that was missing some other updates here https://github.com/opticdev/optic/blob/develop/workspaces/ui-v2/src/optic-components/diffs/render/AskForCommitMessage.tsx#L65

File: workspaces/ui-v2/src/optic-components/hooks/util/useRunOnKeypress.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { KeyboardEvent, useCallback } from 'react';
+
+export const useRunOnKeypress = <T extends (...args: any) => any>(
```
Comment (niclim) [inline workspaces/ui-v2/src/optic-components/hooks/util/useRunOnKeypress.ts:3]: This is something we could use for the future if we wanted to better support power users (keyboard shortcuts to accept, etc)

Comment (devdoshi) [reply workspaces/ui-v2/src/optic-components/hooks/util/useRunOnKeypress.ts:3]: This looks cool, but I would prefer if we stuck to standard web form ux, unless that is way more complicated. I do see the value of supporting future shortcuts, but for now I'd focus on just making sure people don't have to unnecessarily switch between keyboard and mouse throughout the ux. Thoughts?

**Review by devdoshi — APPROVED** [review]
Looks good, let's discuss the 2 threads (native form capabilities, and whether it makes sense to allow editing contributions in the diff ux) in twist

**Pull request merged**